### PR TITLE
Clustered index

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve?

Make TiBigData/flink compatible with `clustered index`.

### What is changed and how it works?
- use `CommonHanle` under clustered index situation.
- fix classCastException when commonHandle with type `Timestamp` and `Date`. https://github.com/tidb-incubator/TiBigData/issues/192
- fix assertException caused by the missing the precision and scale while conversion.
- convert Bit to boolean only when the length of Bit is 1.

### Check List

Tests 

- [X] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Documentation

- [X] enhance TiBigData compatibility

### CODE REVIEW

File: flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/connector/utils/TiDBRowConverter.java [hunk 1]
```diff
@@ -56,7 +56,12 @@ public static org.apache.flink.table.types.DataType toFlinkType(
     org.apache.flink.table.types.DataType flinkType;
     switch (dataType.getType()) {
       case TypeBit:
-        flinkType = DataTypes.BOOLEAN();
+        // Only Convert Bit(1) to Boolean
+        if (dataType.getLength() == 1) {
+          flinkType = DataTypes.BOOLEAN();
+        } else {
+          flinkType = DataTypes.BIGINT();
```
Comment (xuanyu66) [inline flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/connector/utils/TiDBRowConverter.java:63]:
```suggestion
	// TODO  it's better to convert `BIT(n)` to type `Binary(n/8)`
          flinkType = DataTypes.BIGINT();
```

File: flink/flink-base/src/main/java/io/tidb/bigdata/flink/tidb/TypeUtils.java [hunk 2]
```diff
@@ -63,7 +63,12 @@ public static DataType getFlinkType(io.tidb.bigdata.tidb.types.DataType dataType
     int length = (int) dataType.getLength();
     switch (dataType.getType()) {
       case TypeBit:
-        return DataTypes.BOOLEAN();
+        // Only Convert Bit(1) to Boolean
+        if (dataType.getLength() == 1) {
+          return DataTypes.BOOLEAN();
+        } else {
+          return DataTypes.BIGINT();
```
Comment (xuanyu66) [inline flink/flink-base/src/main/java/io/tidb/bigdata/flink/tidb/TypeUtils.java:70]:
```suggestion
	  // TODO : it's better to convert `BIT(n)` to type `Binary(n/8)`
          return DataTypes.BIGINT();
```

**Comment by xuanyu66** [comment]
@humengyu2012 @zhangyangyu @shiyuhang0 PTAL

File: flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/connector/utils/TiDBRowConverter.java [hunk 3]
```diff
@@ -56,7 +56,13 @@ public static org.apache.flink.table.types.DataType toFlinkType(
     org.apache.flink.table.types.DataType flinkType;
     switch (dataType.getType()) {
       case TypeBit:
-        flinkType = DataTypes.BOOLEAN();
+        // Only Convert Bit(1) to Boolean
+        if (dataType.getLength() == 1) {
+          flinkType = DataTypes.BOOLEAN();
+        } else {
+	  // TODO : it's better to convert `BIT(n)` to type `Binary(n/8)`
```
Comment (shiyuhang0) [inline flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/connector/utils/TiDBRowConverter.java:63]: to n/8 may loss precision for  `n%8!=0`.

File: tidb/pom.xml [hunk 4]
```diff
@@ -21,6 +21,7 @@
     <protobuf.version>3.5.1</protobuf.version>
     <grpc.version>1.38.0</grpc.version>
     <proto.folder>${basedir}/proto</proto.folder>
+    <argLine>-Dfile.encoding=UTF-8 -Duser.timezone=GMT+8</argLine>
```
Comment (shiyuhang0) [inline tidb/pom.xml:24]: does specifying a time zone in pom may cause the error for the user of another time zone?

**Review by humengyu2012 — APPROVED** [review]

**Review by shiyuhang0 — APPROVED** [review]

**Pull request merged**